### PR TITLE
Support min_tokens in sgl.gen

### LINKED ISSUE #1564: [Feature] Add `choices` in `/generate` endpoint and add `min_new_tokens` in `sgl.gen()`
### Checklist

- [X] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 2. Please use English, otherwise it will be closed.

### Motivation

Add `choices` in `/generate` endpoint and add `min_new_tokens` in `sgl.gen()`

### Related resources

_No response_

### PR DESCRIPTION
**Pull request by ByronHsu**
## Motivation

Fix https://github.com/sgl-project/sglang/issues/1564. Currently, the frontend (`SglSamplingParams`) and backend (`SamplingParams`) SamplingParams are not consistent. The frontend one does not expose `min_new_tokens`. This PR adds necessary fields to the frontend to enable `min_new_tokens` be passed to the backend 
 
### Context

The lifecycle of sampling params is as followed. The frontend side currently does not expose `min_new_tokens`
JPEG image-4C19-9235-0D-0

### Add `min_tokens` to the frontend (`sgl.gen`, `sgl.gen_int`, `sgl.gen_string`)

Fill `min_tokens` in `api`, `ir`, and `Interpreter`. Enabling `min_tokens` to be serialized into http json object.

### Example

Code:

```python
"""
This example demonstrates how to use `min_tokens` to enforce sgl.gen to generate a longer sequence

Usage:
python3 sgl_gen_min_tokens.py
"""

import sglang as sgl


@sgl.function
def long_answer(s):
    s += sgl.user("What is the capital of the United States?")
    s += sgl.assistant(sgl.gen("answer", min_tokens=64, max_tokens=128))

@sgl.function
def short_answer(s):
    s += sgl.user("What is the capital of the United States?")
    s += sgl.assistant(sgl.gen("answer"))


if __name__ == "__main__":
    runtime = sgl.Runtime(
        model_path="/shared/public/elr-models/meta-llama/Meta-Llama-3.1-8B-Instruct/07eb05b21d191a58c577b4a45982fe0c049d0693/"
    )
    sgl.set_default_backend(runtime)

    state = long_answer.run()
    print("=" * 20)
    print("Longer Answer", state["answer"])

    state = short_answer.run()
    print("=" * 20)
    print("Short Answer", state["answer"])


    runtime.shutdown()
```

Result:

```
====================
Longer Answer The capital of the United States is Washington, D.C. (short for District of Columbia). It's located on the East Coast, along the Potomac River, and serves as the seat of the federal government. The White House, the Capitol Building, and many other important government institutions are all located in Washington, D.C.
====================
Short Answer The capital of the United States is Washington, D.C. (short for District of Columbia).

```
### Test

Add `test_gen_min_new_tokens` in `test_programs` to ensure min_tokens is effective by checking the generated length > min_tokens

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by ByronHsu** [comment]
@zhyncs @hnyls2002 can you take a look at this PR? thanks!

File: python/sglang/test/test_programs.py [hunk 1]
```diff
@@ -517,3 +517,56 @@ def few_shot_hellaswag(s, question, choices):
...
+        s += sgl.user("What is the capital of the United States?")
+        s += sgl.assistant(
+            sgl.gen("answer", min_tokens=MIN_TOKENS, max_tokens=MAX_TOKENS)
+        )
+
+    @sgl.function
+    def convo_2(s):
+        s += sgl.user("What is the most beautiful integar?")
+        s += sgl.assistant(
+            sgl.gen_int("answer", min_tokens=MIN_TOKENS, max_tokens=MAX_TOKENS)
```
Comment (merrymercy) [inline python/sglang/test/test_programs.py]: `gen_int` uses `regex` under the hood. I think there is some conflict between `min_tokens` and `regex`. Do you need to use them together?

Comment (ByronHsu) [reply python/sglang/test/test_programs.py]: I think we can just pass `None` for `min_new_tokens` in `SglGen` for `gen_int` ([line](https://github.com/sgl-project/sglang/pull/1573/files#diff-192192a73b4db4b25f305dc49a35aeb3e86789d78b07c01b589243e2f2da9b18R145))? What about `gen_string`? Do we also want to pass `None`?

Comment (ByronHsu) [reply python/sglang/test/test_programs.py]: had a offline discussion. will set both to `None`

File: test/lang/test_srt_backend.py [hunk 2]
```diff
@@ -69,6 +70,11 @@ def test_hellaswag_select(self):
             accuracy, latency = test_hellaswag_select()
             assert accuracy > 0.71, f"{accuracy=}"
 
+    # TODO (ByronHsu): intentionally add "0" in the test name to ensure the alpha-numeric order is ahead, so this test is run before test_decode_int
+    # See issue at https://github.com/sgl-project/sglang/issues/1575
+    def test_0_gen_min_new_tokens(self):
```
Comment (merrymercy) [inline test/lang/test_srt_backend.py]:
```suggestion
    def test_gen_min_new_tokens(self):
```

File: python/sglang/test/test_programs.py [hunk 3]
```diff
@@ -517,3 +517,35 @@ def few_shot_hellaswag(s, question, choices):
     accuracy = np.mean(np.array(preds) == np.array(labels))
 
     return accuracy, latency
+
+
+def test_gen_min_new_tokens(model_path):
```
Comment (merrymercy) [inline python/sglang/test/test_programs.py]: Use `global_config.default_backend.endpoint.get_model_name()` to get the model name?

Comment (ByronHsu) [reply python/sglang/test/test_programs.py]: cool! didn't know we can do that

**Pull request merged**